services/core-web/common/components/tailings/AssociatedDams.tsx

```diff
@@ -12,40 +12,76 @@ import { getTsf } from "@mds/common/redux/reducers/tailingsReducer";
 import moment from "moment";
 import { storeDam } from "@mds/common/redux/actions/damActions";
 import { useHistory } from "react-router-dom";
-import { EditIcon } from "@/assets/icons";
 import { ADD_DAM, EDIT_DAM } from "@/constants/routes";
-import { IDam, INoticeOfDeparture, ITailingsStorageFacility } from "@mds/common";
+import { IDam, ITailingsStorageFacility } from "@mds/common";
 import { RootState } from "@/App";
 import { ColumnsType } from "antd/lib/table";
 import CoreTable from "@mds/common/components/common/CoreTable";
+import { EDIT_OUTLINE } from "@mds/common/constants/assets";
+import EyeOutlined from "@ant-design/icons/EyeOutlined";
+import { renderActionsColumn } from "@mds/common/components/common/CoreTableCommonColumns";
 
 interface AssociatedDamsProps {
   tsf: ITailingsStorageFacility;
   storeDam: typeof storeDam;
   isCore?: boolean;
+  userAction: string;
+  canEditTSF: boolean;
 }
 
 const AssociatedDams: FC<AssociatedDamsProps> = (props) => {
   const history = useHistory();
-  const { tsf, isCore } = props;
+  const { tsf, isCore, userAction, canEditTSF } = props;
 
-  const handleNavigateToEdit = (event, dam) => {
+  const handleNavigateToEdit = (event, dam, userAction) => {
     event.preventDefault();
     props.storeDam(dam);
     const url = EDIT_DAM.dynamicRoute(
       tsf.mine_guid,
       dam.mine_tailings_storage_facility_guid,
-      dam.dam_guid
+      dam.dam_guid,
+      userAction
     );
     history.push(url);
   };
 
   const handleNavigateToCreate = () => {
     props.storeDam({});
-    const url = ADD_DAM.dynamicRoute(tsf.mine_guid, tsf.mine_tailings_storage_facility_guid);
+    const url = ADD_DAM.dynamicRoute(
+      tsf.mine_guid,
+      tsf.mine_tailings_storage_facility_guid,
+      userAction
+    );
     history.push(url);
   };
 
+  const renderDamActions = () => {
+    let actions = [
+      {
+        key: "edit",
+        label: "Edit Dam",
+        icon: <img src={EDIT_OUTLINE} className="icon-sm padding-sm--right primary-colour" />,
+        clickFunction: (_event, record) => {
+          handleNavigateToEdit(event, record, "edit");
+        },
+      },
+      {
+        key: "view",
+        label: "View Dam",
+        icon: <EyeOutlined className="icon-sm padding-sm--right" />,
+        clickFunction: (_event, record) => {
+          handleNavigateToEdit(event, record, (userAction === "edit" ? "editView" : "view"));
+        },
```

Comment:
What does editView mean? Seems like it would be simpler to just make it "edit".

Reply:
Ah editView was setup because we have a scenario where if a user can edit tsf and dams, they will be able to see the edit and view options in the action menu. If the user with edit access clicks on the dam view button, it will bring them to a dam page with all the user inputs disabled when the user moves back out of the dam page, and into the TailingsSummaryPage we still want them to have the ability to edit things. Having the editView helps let us know the user should still be able to edit

Reply:
Looks like we are already tracking their permissions on the page that consumes it, so I'm not sure that we need this other variable to also indicate that they should have permission to edit. That userAction still really looks to me like it wants to be a boolean- it is always compared to a string literal. A string constant would be better than a string literal, but a boolean I really think would be best. You're either in view mode or edit mode,  and the page itself can track whether the user has permission to change that.

Reply:
I think I should have addressed the points in the latest commit. The string comparisons are converted to be boolean, the route will depend on a boolean to determine view or edit. I tried using `history.goBack()` but found the load time of using that versus `history.push` to go back to previous page to be noticeably longer, so ended up sticking with `history.push`.

Reply:
The boolean change has now been applied.